modules/core/src/main/scala/mapping.scala

```diff
@@ -5,21 +5,43 @@ package edu.gemini.grackle
 
 import cats.Monad
 import cats.implicits._
-import io.circe.Encoder
-
+import io.circe.{Encoder, Json}
 import Query.Select
 import QueryCompiler.{ComponentElaborator, SelectElaborator}
-import QueryInterpreter.mkErrorResult
+import QueryInterpreter.{mkErrorResult, mkOneError}
+import cats.data.NonEmptyChain
 
 trait Mapping[F[_]] {
   implicit val M: Monad[F]
 
   val schema: Schema
   val typeMappings: List[TypeMapping]
-
+  
   def typeMapping(tpe: Type): List[TypeMapping] =
     typeMappings.filter(_.tpe.nominal_=:=(tpe))
 
+  def validate: Either[NonEmptyChain[Json], String] = {
```

Comment:
Ahh ... right, I see what you means about the result type looking odd. Why not return `Chain[Json]` where an empty chain indicates no errors detected?

Reply:
Ok, yes that sounds good.